### PR TITLE
add chapter to describe how to acquire permission

### LINKED ISSUE #40114: [doc] add document to describe how to acquire the permission to push recording
Need to add a chapter to describe what permission is needed and how to acquire it in https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/Azure.Core.TestFramework#recording

### PR DESCRIPTION
**Pull request by chunyu3**
# Contributing to the Azure SDK
Fix https://github.com/Azure/azure-sdk-for-net/issues/40114
- add a description about the permission required to push data to azure-sdk-assets and how to acquire permission

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/core/Azure.Core.TestFramework/README.md [hunk 1]
```diff
@@ -251,6 +251,8 @@ test-proxy push -a <path-to-assets.json>
 
 On completion of the push, a newly created tag will be stamped into the `assets.json` file. This new tag must be committed and pushed to your package directory along with any other changes.
 
+☑️ Permission is required for updating the asserts repository. If you failed in test-proxy push, please [join a partner write team](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/785/Externalizing-Recordings-(Asset-Sync)?anchor=permissions-to-%60azure/azure-sdk-assets%60).
```
Comment (ArcturusZhang) [inline sdk/core/Azure.Core.TestFramework/README.md:254]:
```suggestion
☑️ Permission is required for updating the assets repository. If you failed when executing `test-proxy push` command, please [join a partner write team](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/785/Externalizing-Recordings-(Asset-Sync)?anchor=permissions-to-%60azure/azure-sdk-assets%60).
```

**Review by archerzz — APPROVED** [review]

**Review by lirenhe — APPROVED** [review]

File: sdk/core/Azure.Core.TestFramework/README.md [hunk 2]
```diff
@@ -251,6 +251,8 @@ test-proxy push -a <path-to-assets.json>
 
 On completion of the push, a newly created tag will be stamped into the `assets.json` file. This new tag must be committed and pushed to your package directory along with any other changes.
 
+☑️ Permission is required for updating the assets repository. If you failed when executing `test-proxy push` command, please [join a partner write team](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/785/Externalizing-Recordings-(Asset-Sync)?anchor=permissions-to-%60azure/azure-sdk-assets%60).
```
Comment (jsquire) [inline sdk/core/Azure.Core.TestFramework/README.md:254]: This is great for internal contributors with access to the linked documentation, however, that is not a public link and non-Microsoft contributors cannot access it.   

We really need an approach for this that allows community contributors to update test assets, but I don't believe that one exists at this time.  Assuming that hasn't changed, can we please add a note that this documentation is "Microsoft Internal" and community contributors will need to work with the reviewers for their PR to get assets merged in?

@JoshLove-msft, @scbedd:  Do we have any update here for community contributor support?

**Review by jsquire — CHANGES_REQUESTED** [review]

File: sdk/core/Azure.Core.TestFramework/README.md (hunk 2 above)
Comment (jsquire) [inline sdk/core/Azure.Core.TestFramework/README.md:254]: What's the reason for using this icon?  I don't see this convention used elsewhere in the doc, and this doesn't really highlight and call attention to this, in my opinion.  It seems more appropriate for defining a checklist.

I'd rather see us annotate with **NOTE:** and/or some kind of bit exclamation point.

Comment (chunyu3) [reply sdk/core/Azure.Core.TestFramework/README.md]: Thanks @jsquire I add the notes now. And we may update later if there is a wiki for community contributor. thanks

Comment (chunyu3) [reply sdk/core/Azure.Core.TestFramework/README.md]: Thanks @jsquire I updated to **NOTE**.

**Review by jsquire — APPROVED** [review]
Thanks, @chunyu3!   LGTM

**Review by kinelski — APPROVED** [review]

**Pull request merged**